addons/mail/static/tests/chatter_tests.js

```diff
@@ -819,5 +821,41 @@ QUnit.test('followers widget: do not display follower duplications', function (a
     form.destroy();
 });
 
+QUnit.test('chatter: close chat window using ESCAPE key', function (assert) {
+    assert.expect(2);
+
+    function createParent(params) {
+        var widget = new Widget();
+
+        testUtils.addMockEnvironment(widget, params);
+        return widget;
+    }
+    var messages = [{
+        attachment_ids: [],
+        author_id: ["1", "John Doe"],
+        body: "A message",
+        date: moment("2016-12-20 09:35:40"),
+        displayed_author: "John Doe",
+        id: 1,
+        is_note: false,
+        is_starred: false,
+        model: 'partner',
+        res_id: 2
+    }];
+    var parent = createParent({
+        data: {},
+    });
+
+    var chat_window = new ChatWindow(parent, 1, "user", false, messages, {});
+    chat_window.appendTo($('#qunit-fixture'));
+
+    chat_window.thread.$el.trigger("click");
+    assert.equal(chat_window.$input.is(":focus"), true, "Focused");
+
+    var upKeyEvent = jQuery.Event( "keyup", {which: 27});
+    var close_chat = chat_window.$el.find('.o_composer_input').trigger(upKeyEvent);
+    assert.equal(!chat_window.folded, true, "Closed chat Window");
```

Comment:
I think that you should write assert.strictEqual(chat_window.folded, false, "chat window should be closed")